### PR TITLE
Document `DD_PROFILING_MAX_FRAMES` / `max_frames` setting for Ruby profiler

### PR DESCRIPTION
**Pull request by ivoanjo**
### What does this PR do?

Document the `DD_PROFILING_MAX_FRAMES`/`max_frames` setting in the Ruby profiler troubleshooting section.

### Motivation

This setting is a rare workaround that most customers should never need to use. Hence, I decided to document it as part of the troubleshooting guide, and not elsewhere.

The plan is to also link to this documentation from the Datadog UX.

### Additional Notes

(N/A)

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by alai97** [comment]
Opened https://datadoghq.atlassian.net/browse/DOCS-4629 for an editorial review!

File: content/en/profiler/profiler_troubleshooting.md [hunk 1]
```diff
@@ -217,11 +217,22 @@ Without this flag, profiles for short-lived Resque jobs will be unavailable.
...
 There is a known incompatibility between Ruby 2.7 and older GCC versions (4.8 and below) that impacts the profiler ([upstream Ruby report][6], [`dd-trace-rb` bug report][7]). This can result in the following error message: "Your ddtrace installation is missing support for the Continuous Profiler because compilation of the Ruby VM just-in-time header failed. Your C compiler or Ruby VM just-in-time compiler seem to be broken."
 
-
 To fix this, update your operating system or Docker image so that the GCC version is something more recent than v4.8.
 
 For further help with this issue, [contact support][2] and include the output of running `DD_PROFILING_FAIL_INSTALL_IF_MISSING_EXTENSION=true gem install ddtrace` and the resulting `mkmf.log` file.
 
+## Frames omitted when backtraces are very deep
+
+The Ruby profiler truncates deep backtraces when collecting profiling data. Truncated backtraces are missing some of their caller functions making it impossible to link them to the root call frame. As a result, truncated backtraces are grouped together under a "N frames omitted" frame.
```
Comment (kayayarai) [inline content/en/profiler/profiler_troubleshooting.md:226]:
```suggestion
The Ruby profiler truncates deep backtraces when collecting profiling data. Truncated backtraces are missing some of their caller functions, making it impossible to link them to the root call frame. As a result, truncated backtraces are grouped together under a `N frames omitted` frame.
```

File: content/en/profiler/profiler_troubleshooting.md [hunk 2]
```diff
@@ -217,11 +217,22 @@ Without this flag, profiles for short-lived Resque jobs will be unavailable.
...
+
+The maximum depth can be increased using the `DD_PROFILING_MAX_FRAMES` environment variable, or via code:
```
Comment (kayayarai) [inline content/en/profiler/profiler_troubleshooting.md:228]:
```suggestion
You can increase the maximum depth with the `DD_PROFILING_MAX_FRAMES` environment variable, or in code:
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

**Review by kayayarai — COMMENTED** [review]
oops, approved.

**Comment by ivoanjo** [comment]
Thank you! :)

**Pull request merged**